src/codegen/return_value.rs

```diff
@@ -43,13 +43,29 @@ impl ToReturnValue for library::Parameter {
 impl ToReturnValue for analysis::return_value::Info {
     fn to_return_value(&self, env: &Env, is_trampoline: bool) -> String {
         match self.parameter {
-            Some(ref par) => par.to_return_value(env, is_trampoline),
+            Some(ref par) => {
+                let name = par.to_return_value(env, is_trampoline);
+                if self.nullable_return_is_error.is_some() && name.starts_with(" -> Option<") {
+                    let mut name = name.replace("Option<", "Result<");
```

Comment:
I'm definitely not a big fan of this! `Option<Option<` would be bad. Please change it to:

```rust
let parts = name.splitn(2, "<").collect();
if parts.len() > 1 && parts[0] == "Option" {
    format!("Result<{}", parts[1])
} else {
    parts.join('<')
}
```

Reply:
Or as we already checked for " -> Option<" we can replace it to " -> Result<"

Reply:
That'd be better indeed!

Reply:
@EPashkin you mean simply doing `format!(" -> Result<{}", name[11..]")` or similar?

Reply:
yes, or change replace text